tests/test_content_client.py

```diff
@@ -0,0 +1,169 @@
+    if os_release == "6":
+        katello_client.run_expect([0], 'yum -y install katello-host-tools')
+
+    katello_client.run('yum -y install walrus-0.71')
+
+    host = foreman_host(katello_client, api)
+    errata = api.resource('errata').call('index', {'host_id': host['id']})
+    errata_ids = [erratum['errata_id'] for erratum in errata['results']]
+
+    assert entities['errata_id'] in errata_ids
+
+
+@pytest.mark.katello_agent
+def test_katello_agent_package_install(katello_agent_client, api, os_release):
+    host = foreman_host(katello_agent_client, api)
+    task = api.resource('host_packages').call('install', {'host_id': host['id'], 'packages': ['gorilla']})
+    wait_for_task(task, api)
+
+    gorilla = katello_agent_client.package('gorilla')
+    assert gorilla.is_installed
+
+
+@pytest.mark.katello_agent
+def test_katello_agent_errata_install(katello_agent_client, entities, api, os_release):
+    katello_agent_client.run('yum -y install walrus-0.71')
+    host = foreman_host(katello_agent_client, api)
+    task = api.resource('host_errata').call('apply', {
+        'host_id': host['id'],
+        'errata_ids': [entities['errata_id']],
+        'included': {},
+        'excluded': {}
+    })
+    wait_for_task(task, api)
+
+    walrus = katello_agent_client.package('walrus')
+    assert walrus.is_installed
+    assert walrus.version == '5.21'
+
+
+@pytest.mark.katello_agent
+def test_katello_agent_remove_package(katello_agent_client, api, os_release):
+    host = foreman_host(katello_agent_client, api)
+
+    katello_agent_client.run_expect([0], 'yum -y install gorilla')
+    gorilla = katello_agent_client.package('gorilla')
+    assert gorilla.is_installed
+
+    task = api.resource('host_packages').call('remove', {'host_id': host['id'], 'packages': ['gorilla']})
+    wait_for_task(task, api)
+    gorilla = katello_agent_client.package('gorilla')
+    assert not gorilla.is_installed
+
+
+@pytest.mark.containers
+def test_fetch_container_content(katello_client, entities, registration_hostname, os_release, user):
+    if os_release == "6":
+        pytest.skip("container content not supported on EL6")
+
+    container_label = f"{entities['organization_label']}-{entities['product_label']}-{entities['container_repository_label']}".lower()
+    docker_connection = f"docker://{registration_hostname}/{container_label}"
+
+    katello_client.run('yum -y install skopeo')
+    katello_client.run('mkdir containers')
+    katello_client.run_expect([0], f"curl -O http://{registration_hostname}/pub/katello-server-ca.crt")
+    katello_client.run_expect([0], f"skopeo copy {docker_connection} dir:containers --src-creds {user.username}:{user.password} --src-cert-dir .")
+    katello_client.run_expect([1], f"skopeo copy {docker_connection} dir:containers --src-creds doesnotexist:changeme --src-cert-dir .")
+    katello_client.run_expect([1], f"skopeo copy docker://{registration_hostname}/{container_label} dir:containers --src-cert-dir .")
```

Comment:
I'd swap over these checks: first use invalid credentials, assert that there is nothing in the containers directory, then valid credentials and finally assert that there is content in the containers directory.